packages/smooth_app/lib/pages/product/add_new_product/add_new_product_type.dart

```diff
@@ -0,0 +1,226 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:smooth_app/generic_lib/design_constants.dart';
+import 'package:smooth_app/resources/app_icons.dart' as icons;
+import 'package:smooth_app/themes/smooth_theme_colors.dart';
+import 'package:smooth_app/themes/theme_provider.dart';
+
+class ProductTypeRadioListTile extends StatefulWidget {
+  const ProductTypeRadioListTile({
+    required this.productType,
+    required this.checked,
+    required this.onChanged,
+    super.key,
+  });
+
+  final ProductType productType;
+  final bool checked;
+  final void Function(ProductType) onChanged;
+
+  @override
+  State<ProductTypeRadioListTile> createState() => _AddNewProductType();
+}
+
+class _AddNewProductType extends State<ProductTypeRadioListTile>
```

Comment:
I strongly prefer some coherence among the file name, the class name and the state class name.

Reply:
Nice catch, I didn't see that one